internal/pkg/addon/storage.go

```diff
@@ -112,6 +138,34 @@ func NewS3(input *S3Props) *S3 {
 	}
 }
 
+// MarshalBinary serializes the RDS object into a binary YAML CF template.
+// Implements the encoding.BinaryMarshaler interface.
+func (r *RDS) MarshalBinary() ([]byte, error) {
+	var addonPath string
+	switch r.Engine {
+	case RDSEngineTypeMySQL:
+		addonPath = rdsMySQLAddonPath
+	case RDSEngineTypePostgreSQL:
+		addonPath = rdsPostgreSQLAddonPath
+	default:
+		return nil, errInvalidEngineType
+	}
+	content, err := r.parser.Parse(addonPath, *r, template.WithFuncs(storageTemplateFunctions))
+	if err != nil {
+		return nil, err
+	}
+	return content.Bytes(), nil
+}
+
+// NewRDS creates a new RDS marshaler which can be used to write CF via addonWriter.
+func NewRDS(input *RDSProps) *RDS {
+	return &RDS{
+		RDSProps: *input,
```

Comment:
Maybe just
```suggestion
func NewRDS(input RDSProps) *RDS {
	return &RDS{
		RDSProps: input,
```
to avoid possible null pointer exception.